src/App.tsx

```diff
@@ -15,48 +17,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortField {
+  string = 'alphabetically',
+  num = 'length',
+}
+
+interface GoodsProps {
+  sortField: SortField | '';
+  reversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reversed }: GoodsProps,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.string:
+          return good1.localeCompare(good2);
+
+        case SortField.num:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortField | ''>('');
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    {
+      sortField,
+      reversed,
+    },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortField !== SortField.string },
+          )}
+          onClick={() => setSortField(SortField.string)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortField !== SortField.num },
+          )}
+          onClick={() => setSortField(SortField.num)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': !reversed },
+          )}
+          onClick={() => setReversed(!reversed)}
```

Comment:
better to use [prevState](https://blog.greenroots.info/react-hook-usestate-lazy-initialization-previous-state#heading-how-to-get-the-previous-state-data-and-use-it) for such cases